src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,573 @@
+    #     in_source_only_indirect_mondo_edges, mondo_data_map)
+    #   - Indirect SCRs that exist in Mondo but not in source
+    # in_mondo_only_indirect_mondo_edges: Set[RELATIONSHIP] = rels_indirect_mondo_mondo.difference(
+    #     rels_indirect_source_mondo)
+    # in_mondo_only_indirect: List[Dict] = _edges_with_metadata_from_plain_edges(
+    #     in_mondo_only_indirect_mondo_edges, mondo_data_map)
+
+    # - direct <--> ancestor
+    #   - Direct SCRs in source that don't exist at all in Mondo
+    in_source_direct_not_in_mondo_source_edges = rels_direct_source_source.difference(ancestors_mondo_source)
+    in_source_direct_not_in_mondo: List[Dict] = _edges_with_metadata_from_plain_edges(
+        in_source_direct_not_in_mondo_source_edges, source_data_map)
+    #   - Direct SCRs in Mondo that don't exist at all in source
+    in_mondo_direct_not_in_source_mondo_edges: Set[RELATIONSHIP] = (
+        rels_direct_mondo_mondo.difference(ancestors_source_mondo))
+    # in_mondo_direct_not_in_source: List[Dict] = _edges_with_metadata_from_plain_edges(
+    #     in_mondo_direct_not_in_source_mondo_edges, mondo_data_map)
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
+    # Google doc about cases:
+    # https://docs.google.com/document/d/1H8fJKiKD-L1tfS-2LJu8t0_2YXJ1PQJ_7Zkoj7Vf7xA/edit#heading=h.9hixairfgxa1
+    print('Creating outputs for synchronization cases')
+    common_sort_cols = ['subject_mondo_id', 'object_mondo_id', 'subject_source_id', 'object_source_id']
+
+    # Case 1: SCR direct in source and Mondo
+    df1 = pd.DataFrame(in_both_direct)
+    if len(df1) == 0:
+        df1 = pd.DataFrame(columns=list(ROBOT_SUBHEADER[0].keys()))
+    df1 = df1.sort_values(common_sort_cols)
+    df1 = pd.concat([pd.DataFrame(ROBOT_SUBHEADER), df1])
+    df1.to_csv(outpath_confirmed, sep='\t', index=False)
+
+    # Case 2: SCR is direct in source, but indirect Mondo
+    pass  # no output for this case
+
+    # Case 3: SCR is direct in the source, but not at all in Mondo
+    df3 = pd.DataFrame(in_source_direct_not_in_mondo)
+    if len(df3) == 0:
+        df3 = pd.DataFrame(columns=list(ROBOT_SUBHEADER[0].keys()))
+    # - warning cases
+    #   is.na() checks: only object_mondo_id should ever be missing, but checking both to be safe
+    unmigrated_terms = set(df3[df3['object_mondo_id'].isna()]['object_source_id']).union(
+        set(df3[df3['subject_mondo_id'].isna()]['object_source_id']))
+    if len(unmigrated_terms) >= 200:
+        logging.warning(f'Warning: {len(unmigrated_terms)} unmigrated source terms found in "{ontology_name}"; '
+                        f'higher number than expected.')
+    # - format
+    #   only object_mondo_id should ever be missing, but checking both to be safe
+    df3 = df3[~df3['subject_mondo_id'].isna() & ~df3['object_mondo_id'].isna()].sort_values(common_sort_cols)
+    # - obsolete cases
```

Comment:
## `ordo.subclass.added-obsolete.robot.tsv` changed
I'm not sure why, but after running, this changed. It has sufficient sorting, I think. I didn't think I changed anything in my code that would cause this to be affected. And I didn't refresh any inputs.

I'm not too alarmed but thought you should know.